### ISSUE TITLE
Shape anchors

### ISSUE THREAD

**Issue by dvir** [excalidraw/excalidraw#504]
Often when using the arrow/line tool I anchor it to a shape. To do so with the same precision as I would with a pen and paper, I suggest implementing anchors.

When selecting the arrow/line tool (or perhaps when holding down a key?), visualize pre-defined anchors. They can be positioned where resize anchors are currently:

![image](https://user-images.githubusercontent.com/620099/72851238-1046b480-3c79-11ea-93fa-5db4b1e5171b.png)

Perhaps they will look differently, and be a bit bigger to make it easy to do quickly, and maybe even "magnetic".

I can see reasons to make it so any shape anchors will stick to other shapes, in case one wants to draw things like:

![image](https://user-images.githubusercontent.com/620099/72851515-da560000-3c79-11ea-92ea-46893ca45d52.png)

To summarize:
1. When dragging a shape, if one of the shape's anchors is close to another's, show anchors on both shapes.
2. Anchors are either large enough, or have a radius of being "magnetic".
3. When selecting any shape, before starting to drag we will show anchors of shapes around the pointer.
